src/plugins/workspace/public/components/service_card/workspace_list_card.tsx

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import {
+  EuiPanel,
+  EuiLink,
+  EuiDescriptionList,
+  EuiIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiButtonIcon,
+  EuiSpacer,
+  EuiListGroup,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { Subscription } from 'rxjs';
+import moment from 'moment';
+import _ from 'lodash';
+import { CoreStart, WorkspaceObject } from '../../../../../core/public';
+import { navigateToWorkspaceDetail } from '../utils/workspace';
+
+import { WORKSPACE_CREATE_APP_ID, WORKSPACE_LIST_APP_ID } from '../../../common/constants';
+import { WorkspaceEntry, recentWorkspaceManager } from '../../recent_workspace_manager';
+
+const WORKSPACE_LIST_CARD_DESCRIPTIOIN = i18n.translate('workspace.list.card.descriptionh', {
+  defaultMessage:
+    'Workspaces are dedicated environments for organizing and collaborating on your data, dashboards, and analytics workflows. Each Workspace acts as a self-contained space with its own set of saved objects and access controls.',
+});
+
+const MAX_ITEM_IN_LIST = 5;
+
+export interface WorkspaceListCardProps {
+  core: CoreStart;
+}
+
+export interface WorkspaceListItem {
+  id: string;
+  name: string;
+  time?: string | number;
+}
+
+export interface WorkspaceListCardState {
+  availiableWorkspaces: WorkspaceObject[];
+  filter: string;
+  workspaceList: WorkspaceListItem[];
+  recentWorkspaces: WorkspaceEntry[];
+}
+
+export class WorkspaceListCard extends Component<WorkspaceListCardProps, WorkspaceListCardState> {
+  private workspaceSub?: Subscription;
+  constructor(props: WorkspaceListCardProps) {
+    super(props);
+    this.state = {
+      availiableWorkspaces: [],
+      recentWorkspaces: [],
+      workspaceList: [],
+      filter: 'viewed',
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      recentWorkspaces: recentWorkspaceManager.getRecentWorkspaces() || [],
+    });
```

Comment:
Can we simply init the `recentWorkspaces` in constructor?
```typescript
this.state = {
  availiableWorkspaces: [],
  recentWorkspaces: recentWorkspaceManager.getRecentWorkspaces() || [],
  workspaceList: [],
  filter: 'viewed',
};
```

Reply:
thanks, updated